File: app/src/main/java/com/afollestad/photoaffix/data/Photo.java
package com.afollestad.photoaffix.data;

import android.net.Uri;
import android.support.annotation.NonNull;

import com.afollestad.inquiry.annotations.Column;

import java.io.Serializable;

/** @author Aidan Follestad (afollestad) */
public class Photo implements Serializable {

  @Column public long _id;
  @Column public String _data;
  @Column public long datetaken;

  public Photo() {}

  public Photo(Uri uri) {
    _data = uri.toString();
  }

  @NonNull
  public Uri getUri() {
    Uri uri = Uri.parse(_data);
    if (!uri.toString().startsWith("file://") && !uri.toString().startsWith("content://"))
      uri = Uri.parse(String.format("file://%s", uri.toString()));
    return uri;
  }
}

File: app/src/main/java/com/afollestad/photoaffix/data/Photo.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.photoaffix.data

import android.database.Cursor
import android.net.Uri
import com.afollestad.photoaffix.utils.toUri
import java.io.Serializable

/** @author Aidan Follestad (afollestad) */
data class Photo(
  var id: Long,
  var data: String,
  var dateTaken: Long
) : Serializable {

  companion object {
    fun pull(cursor: Cursor): Photo {
      return Photo(
          id = cursor.getLong(cursor.getColumnIndex("_id")),
          data = cursor.getString(cursor.getColumnIndex("_data")),
          dateTaken = cursor.getLong(cursor.getColumnIndex("datetaken"))
      )
    }
  }

  val uri: Uri
    get() {
      var uri = data.toUri()
      if (!uri.toString().startsWith("file://") &&
          !uri.toString().startsWith("content://")
      ) {
        uri = "file://$uri".toUri()
      }
      return uri
    }
}
